### PR TITLE
Service 6

### LINKED ISSUE #5: Sync with onet
`CreateProtocol`-fix

### LINKED ISSUE #6: improve simulation-examples
The simulation-examples have some shortcomings:
- names `Simulation` and `simulation` are too close
- service-simulation doesn't implement `Node`
- `Node` should show how to retrieve the actual index that is about to be initialized
- simulation doesn't have a test

Closes #5 
Closes #6

### PR DESCRIPTION
**Pull request by ineiti**
Closes #5 
Closes #6

### CODE REVIEW

File: simulation/protocol.go [hunk 1]
```diff
@@ -66,18 +66,21 @@ func (e *Simulation) Setup(dir string, hosts []string) (
 // by the server. Here we call the 'Node'-method of the
 // SimulationBFTree structure which will load the roster- and the
 // tree-structure to speed up the first round.
-func (e *Simulation) Node(config *onet.SimulationConfig) error {
-	return e.SimulationBFTree.Node(config)
+func (s *SimulationProtocol) Node(config *onet.SimulationConfig) error {
+	index, _ := config.Roster.Search(config.Conode.ServerIdentity.ID)
```
Comment (nikkolasg) [inline simulation/protocol.go:70]: Templates set the standards way, let's get an error checking here to inspire people to do the same !

File: simulation/service.go [hunk 2]
```diff
@@ -34,27 +33,37 @@ func NewSimulationService(config string) (onet.Simulation, error) {
...
 	}
 	return sc, nil
 }
 
+// Node can be used to initialize each node before it will be run
+// by the server. Here we call the 'Node'-method of the
+// SimulationBFTree structure which will load the roster- and the
+// tree-structure to speed up the first round.
+func (s *SimulationService) Node(config *onet.SimulationConfig) error {
+	index, _ := config.Roster.Search(config.Conode.ServerIdentity.ID)
```
Comment (nikkolasg) [inline simulation/service.go:52]: Same here.

**Review by nikkolasg — APPROVED** [review]
Ok to merge once the error checking is done.

File: simulation/protocol.go (hunk 1 above)
Comment (ineiti) [reply simulation/protocol.go:70]: The second argument is not an error, but I can check for a negative index, which indicates an error.

File: simulation/service.go (hunk 2 above)
Comment (ineiti) [reply simulation/service.go:52]: The second argument is not an error, but I can check for a negative index, which indicates an error.

**Pull request merged**